src/it/resources/org/checkstyle/suppressionxpathfilter/emptylineseparator/SuppressionXpathRegressionEmptyLineSeparator5.java

```diff
@@ -0,0 +1,22 @@
+package org.checkstyle.suppressionxpathfilter.emptylineseparator;
+
+import java.io.IOException;
+
+public class SuppressionXpathRegressionEmptyLineSeparator5 {
+
+    int foo1() throws Exception {
+        int a = 1;
+        int b = 2;
+        try {
+            if (a != b) {
+                throw new IOException();
+            }
+            // warn
```

Comment:
violation should be on last known token that present in AST , it is line 13 `}`.

Reply:
Sir, why we don't consider the `SINGLE_LINE_COMMENT` as last token in AST before empty line. According to AST this is the last token. Section of AST:
```
    |       |   |   |--LITERAL_IF -> if [11:12]
    |       |   |   |   |--LPAREN -> ( [11:15]
    |       |   |   |   |--EXPR -> EXPR [11:18]
    |       |   |   |   |   `--NOT_EQUAL -> != [11:18]
    |       |   |   |   |       |--IDENT -> a [11:16]
    |       |   |   |   |       `--IDENT -> b [11:21]
    |       |   |   |   |--RPAREN -> ) [11:22]
    |       |   |   |   `--SLIST -> { [11:24]
    |       |   |   |       |--LITERAL_THROW -> throw [12:16]
    |       |   |   |       |   |--EXPR -> EXPR [12:22]
    |       |   |   |       |   |   `--LITERAL_NEW -> new [12:22]
    |       |   |   |       |   |       |--IDENT -> IOException [12:26]
    |       |   |   |       |   |       |--LPAREN -> ( [12:37]
    |       |   |   |       |   |       |--ELIST -> ELIST [12:38]
    |       |   |   |       |   |       `--RPAREN -> ) [12:38]
    |       |   |   |       |   `--SEMI -> ; [12:39]
    |       |   |   |       `--RCURLY -> } [13:12]
    |       |   |   |--SINGLE_LINE_COMMENT -> // [14:12]
    |       |   |   |   `--COMMENT_CONTENT ->  warn\n [14:14]   // last token
    |       |   |   `--RCURLY -> } [17:8]                       // empty line before this

```

Reply:
you are right, I recall this.